### PR TITLE
Fix most current minor issues

### PR DESCRIPTION
**Pull request by Minecrell**
Fixes all current checkstyle warnings in the API and sets it to break the build when violations are found so we can keep the master branch clean from new warnings in the future.

I've also updated the IDE code style configurations to be more exact to the Google style and added a cleanup profile for Eclipse.

### CODE REVIEW

File: src/main/java/org/spongepowered/api/block/BlockProperty.java [hunk 1]
```diff
@@ -82,19 +82,19 @@
     Optional<T> getValueForName(String name);
 
     // Subinterface markers for vanilla property types
-    public interface BooleanProperty extends BlockProperty<Boolean> {
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/block/BlockProperty.java]: Why are these not public?

File: src/main/java/org/spongepowered/api/event/block/data/FurnaceSmeltItemEvent.java [hunk 2]
```diff
@@ -29,7 +29,7 @@
 import org.spongepowered.api.util.event.Cancellable;
 
 /**
- * An event that occurs when a {@link Furnace}
+ * An event that occurs when a {@link Furnace} smelt an item.
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/event/block/data/FurnaceSmeltItemEvent.java]: "smelt" => "smelts"

File: src/main/java/org/spongepowered/api/service/permission/PermissionService.java [hunk 3]
```diff
@@ -36,8 +36,8 @@
  */
 public interface PermissionService {
 
-    static final String SUBJECTS_USER = "user";
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/service/permission/PermissionService.java]: Why aren't these `static final`?

File: src/main/java/org/spongepowered/api/block/BlockProperty.java (hunk 1 above)
Comment (Minecrell) [inline src/main/java/org/spongepowered/api/block/BlockProperty.java]: `public` is redundant in interfaces

File: src/main/java/org/spongepowered/api/service/permission/PermissionService.java (hunk 3 above)
Comment (Minecrell) [inline src/main/java/org/spongepowered/api/service/permission/PermissionService.java]: `static final` is redundant in interfaces

**Comment by Minecrell** [comment]
@Zidane Can you take a look at this? (Especially the change to break the build on violations in the API)

**Comment by Zidane** [comment]
@Minecrell 

I'll probably bitch and moan at you later for this but its a solid change solely to keep our code clean. I'll allow it.

**Comment by Minecrell** [comment]
Merged

**Pull request closed without merging**